functions/venue.js

```diff
@@ -333,102 +333,115 @@ exports.updateVenue = functions.https.onCall(async (data, context) => {
-      }
-      if (data.parentId) {
-        updated.parentId = data.parentId;
-      }
+  if (!doc || !doc.exists) {
+    throw new HttpsError("not-found", `Venue ${venueId} not found`);
+  }
 
-      if (data.columns) {
-        updated.columns = data.columns;
-      }
-      if (data.roomVisibility) {
-        updated.roomVisibility = data.roomVisibility;
-      }
+  const updated = doc.data();
 
-      updated.showLiveSchedule = data.showLiveSchedule;
-      updated.showGrid = data.showGrid;
-
-      switch (updated.template) {
-        case VenueTemplate.jazzbar:
-        case VenueTemplate.performancevenue:
-        case VenueTemplate.artpiece:
-        case VenueTemplate.audience:
-          if (data.iframeUrl) {
-            updated.iframeUrl = data.iframeUrl;
-          }
-          break;
-        case VenueTemplate.zoomroom:
-        case VenueTemplate.artcar:
-          if (data.zoomUrl) {
-            updated.zoomUrl = data.zoomUrl;
-          }
-          break;
+  if (data.bannerImageUrl || data.subtitle || data.description) {
+    if (!updated.config) {
+      updated.config = {};
+    }
+    if (!updated.config.landingPageConfig) {
+      updated.config.landingPageConfig = {};
+    }
+  }
+
+  if (data.bannerImageUrl) {
+    updated.config.landingPageConfig.bannerImageUrl = data.bannerImageUrl;
+  }
+
+  if (data.subtitle) {
+    updated.config.landingPageConfig.subtitle = data.subtitle;
+  }
+
+  if (data.description) {
+    updated.config.landingPageConfig.description = data.description;
+  }
+
+  if (data.primaryColor) {
+    if (!updated.theme) {
+      updated.theme = {};
+    }
+    updated.theme.primaryColor = data.primaryColor;
+  }
+
+  if (data.logoImageUrl) {
+    if (!updated.host) {
+      updated.host = {};
+    }
+    updated.host.icon = data.logoImageUrl;
+  }
+
+  if (data.profileQuestions) {
+    updated.profileQuestions = data.profileQuestions;
+  }
+
+  if (data.mapIconImageUrl) {
+    updated.mapIconImageUrl = data.mapIconImageUrl;
+  }
+
+  if (data.mapBackgroundImageUrl) {
+    updated.mapBackgroundImageUrl = data.mapBackgroundImageUrl;
+  }
+
+  if (
+    !data.placement.state ||
+    data.placement.state === PlacementState.SelfPlaced
+  ) {
+    updated.placement = {
+      ...data.placement,
+      state: PlacementState.SelfPlaced,
+    };
+  } else if (data.placementRequests) {
+    updated.placementRequests = data.placementRequests;
+  }
+
+  if (data.bannerMessage) {
+    updated.bannerMessage = data.bannerMessage;
+  }
+
+  if (data.parentId) {
+    updated.parentId = data.parentId;
+  }
+
+  if (data.columns) {
+    updated.columns = data.columns;
+  }
+
+  if (data.roomVisibility) {
+    updated.roomVisibility = data.roomVisibility;
+  }
+
+  if (typeof data.showLiveSchedule === "boolean") {
+    updated.showLiveSchedule = data.showLiveSchedule;
+  }
+
+  if (typeof data.showGrid === "boolean") {
+    updated.showGrid = data.showGrid;
+  }
+
+  switch (updated.template) {
```

Comment:
We could also potentially move these within the `switch` statement, so that they are only handled for the correct `VenueTemplate` types.

Reply:
The direction I think it makes sense to move is more toward template agnostic here, so the less fussy the fields being updated are about the template, the better. But when I rip through this to make the venue config template agnostic, this will all change anyway so your call. I’m ok with it as is.

Reply:
Works for me :)

:shipit: !